test/build.js

```diff
@@ -0,0 +1,271 @@
+const test = require('ava');
+const brunch = require('../lib');
+const { prepareTestDir, teardownTestDir, fileContains, fileNotContains, fileExists } = require('./_test_helper');
+const fixturify = require('fixturify');
+
+test.beforeEach(() => {
+  teardownTestDir();
+  prepareTestDir();
+});
+
+test.serial.cb('basic build', t => {
+  fixturify.writeSync('.', {
+    'brunch-config.js': `module.exports = {
+      files: {
+        javascripts: {
+          joinTo: 'app.js'
+        }
+      }
+    };`,
+    'app': {
+      'assets': {
+        'index.html': '<h1>hello world</h1>'
+      },
+      'initialize.js': 'console.log("hello world")'
+    }
+  });
+
+  brunch.build({}, () => {
+    fileExists(t, 'public/app.js.map');
+    fileContains(t, 'public/app.js', '//# sourceMappingURL=app.js.map');
+    fileContains(t, 'public/app.js', `
+require.register("initialize.js", function(exports, require, module) {
+console.log("hello world")
+});`);
+
+    fileContains(t, 'public/index.html', '<h1>hello world</h1>');
+    t.end();
+  });
+});
+
+test.serial.cb('basic file joining', t => {
+  fixturify.writeSync('.', {
+    'brunch-config.js': `module.exports = {
+      files: {
+        javascripts: {
+          joinTo: 'app.js'
+        }
+      }
+    };`,
+    'app': {
+      'assets': {
+        'index.html': '<h1>hello world</h1>'
+      },
+      'a.js': 'filea',
+      'b.js': 'fileb',
+      'c.js': 'filec'
+    }
+  });
+
+  brunch.build({}, () => {
+    fileExists(t, 'public/app.js.map');
+    fileContains(t, 'public/index.html', '<h1>hello world</h1>');
+    fileContains(t, 'public/app.js', `require.register("a.js", function(exports, require, module) {
+filea
+});
+
+;require.register("b.js", function(exports, require, module) {
+fileb
+});
+
+;require.register("c.js", function(exports, require, module) {
+filec
+});
+`);
```

Comment:
This is probably ok, for now, but maybe we shouldn't heavily rely on the exact contents of output file. What will probably make sense further down the line, something that I've been thinking about implementing when I'll get my hands down to improving our testing, is running the resulting js in a headless webkit and very that things indeed work properly (which will also guard us against bugs in commonjs runtime, etc)

But as I said, totally fine for now